lib/internal/modules/esm/resolve.js

```diff
@@ -230,7 +230,7 @@ function finalizeResolution(resolved, base, preserveSymlinks) {
       fileURLToPath(base));
   }
 
-  const path = fileURLToPath(resolved);
+  const path = fileURLToPath(resolved, base);
```

Comment:
We don't have to set the properties in `internal/errors`, we could set it there, avoiding the addition of arguments on `fileURLToPath`:
```suggestion
  let path;
  try {
    path = fileURLToPath(resolved);
  } catch (err) {
    setOwnProperty(err, 'input', `${resolved}`);
    setOwnProperty(this, 'module', `${base}`);
    throw err
  }
```